server/src/main/java/com/cloud/user/PasswordPolicyImpl.java

```diff
@@ -188,12 +188,12 @@ protected void validateIfPasswordMatchesRegex(String password, String username,
         logger.trace(String.format("Validating if the new password for user [%s] matches regex [%s] defined in the configuration [%s].",
                 username, passwordPolicyRegex, PasswordPolicyRegex.key()));
 
-        if (passwordPolicyRegex == null){
+        if (StringUtils.isBlank(passwordPolicyRegex)) {
             logger.trace(String.format("Regex is null; therefore, we will not validate if the new password matches with regex for user [%s].", username));
             return;
         }
 
-        if (!password.matches(passwordPolicyRegex)){
+        if (!password.matches(passwordPolicyRegex)) {
```

Comment:
Hi @JoaoJandre Does the _password.policy.regex_ setting update checks the existing users password match (at sometime, may be login), and enforces user with non matching password to change the password immediately or after certain grace period?

Reply:
@sureshanaparti All the password policies are applied only on user/account creation or password change. They are not validated on log in.

Reply:
> @sureshanaparti All the password policies are applied only on user/account creation or password change. They are not validated on log in.

ok, thanks for the update @JoaoJandre